inspirehep/modules/records/receivers.py

```diff
@@ -206,3 +215,45 @@ def enhance_after_index(sender, json, record, *args, **kwargs):
 
     elif is_data(json):
         populate_citations_count(record=record, json=json)
+
+
+@after_record_insert.connect
+@after_record_update.connect
+def after_update(sender, *args, **kwargs):
+    json = kwargs.get('record')
+
+    if json.get('deleted'):
+        # in this way the references diff will be the list of references itself
+        record = InspireRecord(json)
+    else:
+        pid_type = get_pid_type_from_schema(json['$schema'])
+        pid_value = json['control_number']
+        try:
+            rec_uuid = PersistentIdentifier.get(pid_type, pid_value).object_uuid
+            record = InspireRecord.get_record(rec_uuid)
+        except NoResultFound:
+            # deleted record
+            record = InspireRecord(json)
+
+    modified_references = record.get_modified_references()
+
+    pids = [
+        str(get_recid_from_ref(ref['record'])) for ref in modified_references
+    ]
+    uuids = [
+        t[0] for t in db.session.query(PersistentIdentifier.object_uuid)
+        .filter(PersistentIdentifier.pid_type == 'lit')
+        .filter(PersistentIdentifier.pid_value.in_(pids))
+        .all()
+    ]
+
+    timeout = current_app.config.get('INDEXER_BULK_REQUEST_TIMEOUT')
+
+    if uuids:
+        return batch_reindex.apply_async(
+            kwargs={
+                'uuids': uuids,
+                'request_timeout': timeout,
+            },
+            queue='indexer_task',
```

Comment:
**Not for this PR** similar question, why does the caller need to know the queue? we should use [routers](http://docs.celeryproject.org/en/latest/userguide/routing.html#routers) in the config instead.